packages/batch/src/types.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Types for batch processing utility
+ */
+import { DynamoDBRecord, KinesisStreamRecord, SQSRecord } from 'aws-lambda';
+
+// types from base.py
+type EventSourceDataClassTypes =
```

Comment:
Let's rename this to `EventSourceTypes`. 

The "data class" nomenclature is specific to Python and doesn't add much to JS/TS developers.